File: summary/dart/11_prime_factorization.dart
import 'dart:io';

List<int> primeFactors(int n) {
  List<int> factors = [];
  int divisor = 2;

  while (n > 1) {
    while (n % divisor == 0) {
      factors.add(divisor);
      n ~/= divisor;
    }
    divisor += 1;
  }

  return factors;
}

void main() {
  print("Enter a number:");
  String? input = stdin.readLineSync();
  if (input != null) {
    int number = int.parse(input);
    List<int> factors = primeFactors(number);
    print("Prime factors: $factors");
  }
}

File: summary/python/11_prime_factorization.py
def prime_factors(n):
    factors = []
    divisor = 2

    while n > 1:
        while n % divisor == 0:
            factors.append(divisor)
            n //= divisor
        divisor += 1

    return factors

# 入力を受け取る
n = int(input())

# 素因数分解を実行
result = prime_factors(n)

# 結果を出力
print(f"素因数分解結果: {result}")
